src/lib/data/langs/langs.js

```diff
@@ -1,11 +1,46 @@
 import LangsList from '@/lib/data/langs/langs-list.json'
 import L10nSingleton from '@/lib/l10n/l10n-singleton.js'
 
+let allLangs = []
+const rtlLangs = []

```

Comment:
I don't agree here - because `rtlLangs` is a small array with maximum of 12 elements (amount of rtl langs).
It has a simple list of languages and nothing else.
There is no information for the object or Map here.

it looks like the following : [ 'ara', 'per', 'syr', .... ] - and that's all.
I understand your logic and could find a usage for such objects (and use them for detection dictionary) - but here it is redundant from my point of view.
I don't think that we should demonize `indexOf` and `includes` methods of the array and don't ever use it .

Reply:
Those explanations make sense, and I won't let this hold up approving the PR. But I still disagree about both points and I'd like to explain why below:

> `rtlLangs` is a small array with maximum of 12 elements

I think it is misleading to say that `rtLangs` is an array of 12 elements when talking about the code in this file. The array is not defined in this file, but calculated from `@/lib/data/langs/langs-list.json`. If the performance of the code in this file depends on the contents of that other file, the code is [tightly coupled](https://en.wikipedia.org/wiki/Coupling_(computer_programming)). This means that if `langs-list.json` changes, the performance of the code could change drastically. It also means that the developer or maintainer of this code has to be aware of the details of `langs-list.json`, increasing cognitive load.

> I don't think that we should demonize `indexOf` and `includes` methods of the array and don't ever use it.

Code largely serves two purposes: 1. it is a way of communicating with the machine and 2. it is a way to communicate your intention to other programmers. Even if there are no performance issues, the type of data structure communicates to other developers what the data structure is intended to be used for. In this case, I suggest using a `Map` because `rtlLangs` is used only to detect membership. Any programmer looking at the code will understand its intended use immediately if a `Map` is used. But a programmer will have to understand the internal workings of the rest of the code in this file to understand its use if an `array` is used.